src/web/client/common/remoteServiceProvider.ts

```diff
@@ -0,0 +1,147 @@
+        const data = await response.json();
+        switch (entity) {
+            case 'adx_webpages':
+                createsinglewebpage(data, entity, queryParamsMap, entitiesSchemaMap, languageIdCodeMap, portalFs);
+                break;
+
+            case 'adx_webfiles':
+                createwebfile(data, entity, portalFs);
+                break;
+
+            case 'adx_webtemplates':
+                createwebtemplate(data, entity, portalFs);
+                break;
+            default:
+                vscode.window.showInformationMessage('None of the entity matched')
+        }
+        vscode.workspace.onDidSaveTextDocument(async (e) => {
+            vscode.window.showInformationMessage('saving file: ' + e.uri);
+            const newFileData = portalFs.readFile(e.uri);
+            const patchRequestUrl = getSaveURLSingleEntity(dataverseOrg, entity, entityId, SINGLE_ENTITY_SAVE_URL_KEY);
+            vscode.window.showInformationMessage(patchRequestUrl)
+            await saveData(accessToken, patchRequestUrl, e.uri, entity, new TextDecoder('utf-8').decode(newFileData));
+        });
+
+    } catch (e: any) {
+        if (e.message.includes('Unauthorized')) {
+            vscode.window.showErrorMessage('Failed to authenticate');
+        }
+    }
+}
+
+function getRequestURLSingleEntity(dataverseOrg: any, entity: string, entityId: string, urlquery: string, entitiesSchemaMap: any) {
+    const parameterizedUrl = dataSourcePropertiesMap.get(urlquery) as string;
+    const requestUrl = parameterizedUrl.replace('{dataverseOrg}', dataverseOrg).replace('{entity}', entity).replace('{entityId}', entityId).replace('{api}', dataSourcePropertiesMap.get('api')).replace('{data}', dataSourcePropertiesMap.get('data')).replace('{version}', dataSourcePropertiesMap.get('version'));
+    return requestUrl + entitiesSchemaMap.get(pathparam_schemaMap.get(entity)).get('_query');
+}
+
+function getSaveURLSingleEntity(dataverseOrg: any, entity: string, entityId: string, urlquery: string) {
+    const parameterizedUrl = dataSourcePropertiesMap.get(urlquery) as string;
+    const requestUrl = parameterizedUrl.replace('{dataverseOrg}', dataverseOrg).replace('{entity}', entity).replace('{entityId}', entityId).replace('{api}', dataSourcePropertiesMap.get('api')).replace('{data}', dataSourcePropertiesMap.get('data')).replace('{version}', dataSourcePropertiesMap.get('version'));
+    return requestUrl;
+}
+function createsinglewebpage(res: any, entity: string, queryParamsMap: any, entitiesSchemaMap: any, languageIdCodeMap: any, portalsFS: PortalsFS) {
+    portalsFS.createDirectory(vscode.Uri.parse(`${PORTALSURISCHEME}:/${PORTALSFOLDERNAME}/${WEBPAGES_FILENAME}/`, true));
+    let languageCode;
+    if (languageIdCodeMap.size) {
+        languageCode = languageIdCodeMap.get(queryParamsMap.get('websiteId')) ? languageIdCodeMap.get(queryParamsMap.get('websiteId')) : DEFAULT_LANGUAGE_CODE;
+    }
+    const adxCopy = res['adx_copy'] ? res['adx_copy'] : ' ';
```

Comment:
These are hardcoded, which files do we have to create should come from the entity 'attributes' in the schema.